nilearn/glm/first_level/first_level.py

```diff
@@ -362,6 +362,9 @@ def fit(self, run_imgs, events=None, confounds=None,
             takes precedence over events and confounds.
 
         """
+        # Raise a warning if both design_matrices and confounds are provided
+        if design_matrices is not None and confounds is not None:
```

Comment:
```suggestion
        if design_matrices is not None and (confounds is not None or events is not None):
```
Based on the warning, it seems like both confounds and events should be checked here.